### PR TITLE
task(settings,content,auth): Add account uid to gling pings

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We can decorate glean pings with raw uid

## This pull request

- Adds account uid (when available) to glean pings

## Issue that this pull request solves

Closes: FXA-10321

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Asides from tests, I was able to manually validate `account_uid` was added to pings.

### CODE REVIEW

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 1]
```diff
@@ -3,6 +3,21 @@
 $schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
 
 account:
+  uid:
```
Comment (dschom) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:6]: At first I tried calling this `user_id` (to be consistent with the existing`user_id_sha256`), but glean-generate threw an error about the name being invalid.

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 2]
```diff
@@ -3,6 +3,21 @@
 $schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
```
Comment (dschom) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:3]: Not sure if subplat-backend should also be updated, but I went ahead and added for consistency.

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 3]
```diff
@@ -3,6 +3,21 @@
...
+      - subplat-team@mozilla.com
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXA-7531
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=tbd
+    expires: never
+    data_sensitivity:
+      - interaction
+
   user_id_sha256:
```
Comment (dschom) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:21]: @chenba Remove this? Or leave it?

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml (hunk 1 above)
Comment (chenba) [reply libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:6]: ooh this one I know..  add
```yaml
    no_lint:
      - COMMON_PREFIX
```
to ignore that error.

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml (hunk 2 above)
Comment (chenba) [reply libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:3]: @StaberindeZA

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 4]
```diff
@@ -3,6 +3,21 @@
 $schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
 
 account:
+  uid:
+    type: string
+    description: |
+      The account's uid
```
Comment (chenba) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:9]: I think for these description we should start writing them for people outside of FxA.  This is known as the firefox/mozilla account id to some.

**Review by chenba — APPROVED** [review]
LGTM.  Let's use `user_id` like you originally planned.  Thanks!

**Pull request merged**